From 50d472cc8caf6f16ac8ce52818143b30d100fa6a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Thu, 10 Jan 2019 12:46:57 +0200
Subject: [PATCH 418/741] crypto: caam/qi2 - fix enqueuing from all cores
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit e00b178677da10b8feb5a4754076f0507e971ed6 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Changing the code to use affine DPIOs had a side effect of allowing all
cores to enqueue jobs towards the crypto accelerator.

The case when number of (Rx, Tx) queue pairs is smaller than number of
cores is not working correctly, since per-core private data structures
are not filled for all cores.

Going forward, allow for all cores to enqueue, but take care of data
structures setup.

Fixes: 27f75fe03c19 ("crypto: caam/qi2 - use affine DPIOs")
Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
index 1a9f93c27c5a..3ae1621f1d9c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
@@ -592,7 +592,7 @@
 				 <&clockgen 4 1>;
 		};
 
-                usb_aux_bus: usb_aux_bus {
+		aux_bus: aux_bus {
                        #address-cells = <2>;
                        #size-cells = <2>;
                        compatible = "simple-bus";
@@ -631,15 +631,15 @@
                                usb3-lpm-capable;
                                snps,dis-u1u2-when-u3-quirk;
                        };
-                };
 
-		sata: sata@3200000 {
-			compatible = "fsl,ls1046a-ahci";
-			reg = <0x0 0x3200000 0x0 0x10000>,
-				<0x0 0x20140520 0x0 0x4>;
-			reg-names = "ahci", "sata-ecc";
-			interrupts = <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clockgen 4 1>;
+                       sata: sata@3200000 {
+                               compatible = "fsl,ls1046a-ahci";
+                               reg = <0x0 0x3200000 0x0 0x10000>,
+                                       <0x0 0x20140520 0x0 0x4>;
+                               reg-names = "ahci", "sata-ecc";
+                               interrupts = <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
+                               clocks = <&clockgen 4 1>;
+                       };
 		};
 
 		qdma: qdma@8380000 {
-- 
2.17.1

